packages/scandipwa/src/route/Checkout/Checkout.container.js

```diff
@@ -301,6 +310,16 @@ export class CheckoutContainer extends PureComponent {
         history.goBack();
     }
 
+    checkIfDownloadableInCart() {
+        const { totals: { items } } = this.props;
+
+        const isDownloadable = items.find(({ product }) => product.type_id === DOWNLOADABLE);
+
+        if (isDownloadable) {
+            history.push(appendWithStoreCode('/account/login'));
+        }
+    }
```

Comment:
The function that checks something should return the result of the check, without any side effects.
This function has a huge side effect of redirecting a user to some other page. Please, rename it to something that'll reflect the purpose more.

Reply:
Also, you might want to show some notification here, for UX reasons.

Reply:
Added notification. There is no notification on Luma theme in this case, so I suggested that it can be like this:
'Please sign in or remove downloadable products from cart!'